File: defects-explorer/src/main/java/cn/edu/sustc/cse/sql/defectsexplorer/config/CorsConfig.java
package cn.edu.sustc.cse.sql.defectsexplorer.config;

import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.CorsRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;

@Configuration
public class CorsConfig extends WebMvcConfigurerAdapter {
    @Override
    public void addCorsMappings(CorsRegistry registry){
        registry.addMapping("/api/**").allowedOrigins("http://localhost:4200");
    }
}

File: defects-explorer/src/main/kotlin/cn/edu/sustc/cse/sql/defectsexplorer/config/CorsConfig.kt
package cn.edu.sustc.cse.sql.defectsexplorer.config

import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.CorsRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter

@Configuration
class CorsConfig : WebMvcConfigurerAdapter() {
    override fun addCorsMappings(registry: CorsRegistry) {
        registry
                .addMapping("/api/**")
                .allowedOrigins("http://localhost:3000")
                .allowedMethods("GET", "POST", "PUT", "DELETE")
    }
}
